docs/advanced/mixin/3.inject.md

```diff
@@ -0,0 +1,133 @@
+
+
+
+### Injection Codes
+Injection codes are the codes in the injection methods that will be injected into the target methods.
+
+Most time you can complete the injection codes as if they were in the corresponding places of target methods. But to access fields or invoke methods, you should use [Shadow][shadow]. 
+
+Additionally,  ```this``` cannot be directly used to refer to the target instance as the codes are in the mixin class. You should cast ```this``` to ```Object``` and then to the target class. It may seem strange elsewhere, but it is normal in Mixin.
+
+```java
+TargetClass self = (TargetClass) (Object) this;
+```
+
+If you want to end the methods in advance, or "return" the target method in your injection codes, you should add an optional parameter ```cancellable``` to the annotation.
+
+```java
+@Inject(method = {""}, at = {@At("")}, cancellable = true)
+```
+
+Then use ```CallbackInfo#cancel()``` to end the target method if it is void
+or use ```CallbackInfoReturnable#setReturnValue(T)``` to set the return value and end the target method.
+
+Advanced
+--------
+
+### Method Overloading
+
+Methods with different parameter types can have the same name. This is called method overloading, such as ```methodName(int)``` and ```methodName(int,int)```
+
+To inject into overloaded methods, you should explicitly designate the method name as well as their descriptors in the ```()```. The following [Oracle's document][doc] can help you to finish this. Accordingly, the parameters of their injection methods are also different.
+
+| FieldType term | Type      | Interpretation                                                                    |
+|----------------|-----------|-----------------------------------------------------------------------------------|  
+| B              | byte      | signed byte                                                                       |
+| C              | char      | Unicode character code point in the Basic Multilingual Plane, encoded with UTF-16 |
+| D              | double    | double-precision floating-point value                                             |
+| F              | float     | single-precision floating-point value                                             |
+| I              | int       | int	integer                                                                      |
+| J              | long      | long integer                                                                      |
+| L ClassName ;  | reference | an instance of class ClassName                                                    |
+| S              | short     | signed short                                                                      |
+| Z              | boolean   | true or false                                                                     |
+| [              | reference | one array dimension                                                               |
```

Comment:
This is already in the AT docs correct? This should probably be moved into its own section then.